blockchain/fullblocktests/generate.go

```diff
@@ -575,7 +579,22 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	//   incorrectly ordered txouts, scripts that do not involve p2pkh or
 	//   p2sh addresses, etc
 	// - Try to vote with an immature ticket
-	// - Try to vote before stake validation height
+
+	// Create block that tries to vote before stake validation height.
+	//   bse#
+	//       \ -> bis1
+	g.NextBlock("bis1", nil, nil, func(b *wire.MsgBlock) {
+		// block bse0 being the first block to purchase tickets will be the
+		// first block with mature tickets.
+		voteBlock := g.BlockByName("bse0")
+		ticketTx := voteBlock.STransactions[1]
+		g.CreateVoteTx(voteBlock, ticketTx, ticketTx.TxIn[0].BlockHeight,
```

Comment:
This should be voting on the previous block `bseTipName`, not `bse0` which would not be a valid vote even if the chain were after SVH at this point.  Also, in this block, the tickets from `bse1` would be becoming valid, which would mean it would be necessary to figure out which of the 10 possible tickets are the winning ones to ensure that the vote would otherwise be valid if it weren't for not being over SVH.

Recall that the intent of these tests is to ensure everything else about the block is valid except for the very specific failure condition being tested.

The way to fix this is to generate up to the block _before_ the first block in which tickets become live (e.g. change `liveTicketHeight := stakeEnabledHeight + 3` to `beforeLiveTickets := stakeEnabledHeight + 2` so that this block is the one in which the first 5 tickets become live, and then move the `g.AssertLiveTickets()` to after this block is generated and before the call to `rejected`.

In fact, I would change the assertion to assert a specific number of tickets are live, and specifically assert there are 5 live tickets.  e.g.

```Go
// AssertPoolSize panics if the current tip block associated with the generator
// does not indicate the specified pool size.
func (g *Generator) AssertPoolSize(expected uint32) {
	poolSize := g.tip.Header.PoolSize
	if poolSize != expected {
		panic(fmt.Sprintf("pool size for block %q is %d instead of expected "+
			"%d", g.tipName, poolSize, expected))
	}
}

...
	g.NextBlock("bis1", nil, nil, func(b *wire.MsgBlock) {
		// block bse0 being the first block to purchase tickets will be the
		// first block with mature tickets.
		ticketBlock := g.BlockByName("bse0")
		voteBlock := g.BlockByName(bseTipName)
		ticketTx := ticketBlock.STransactions[1]
		g.CreateVoteTx(voteBlock, ticketTx, ticketTx.TxIn[0].BlockHeight,
			ticketTx.TxIn[0].BlockIndex)
		b.AddSTransaction(ticketTx)
		b.Header.Voters++
	})
	g.AssertPoolSize(5)
	rejected(blockchain.ErrInvalidEarlyStakeTx)
	g.SetTip(bseTipName)
```